### ISSUE TITLE
Primefaces Push: Add the ability to set org.atmosphere.cpr.asyncSupport in web.xml

### ISSUE THREAD

**Issue by miron4dev** [primefaces/primefaces#840]
Continuation of issue [PF 5.0+ Push not working if init-param asyncSupport used in web.xml](https://code.google.com/p/primefaces/issues/detail?id=7526)

I can't set org.atmosphere.cpr.asyncSupport parameter in web.xml.

Next exception occurred in such case:
`No AtmosphereHandler maps request for /testprimepush/primepush/counter with mapping {/counter=AtmosphereHandlerWrapper{ atmosphereHandler=PushEndpointHandlerProxy proxy for fr.gmf.testpush.counter.CounterResource, broadcaster=
    Name: /counter
    AtmosphereResource: 0
    BroadcasterCache org.atmosphere.cache.UUIDBroadcasterCache }}`

Why it's necessary? If you use JBoss EAP 6.2, Atmosphere resolves JBossWebCometSupport as AsyncSupport implementation. JBoss EAP doesn't include APR by default and next exception occur every application redeploy. It's not affect functionality, but such stacktrace is not ok.

`
[JBossWebCometSupport] [HttpEvent is null, JBoss APR Not Properly installed]
 ]] [http-127.0.0.1/127.0.0.1:8080-1]
[AtmosphereFramework] [Failed using comet support: org.atmosphere.container.JBossWebCometSupport, error: JBoss failed to detect this is a Comet application because the APR Connector is not enabled.  ]] [http-127.0.0.1/127.0.0.1:8080-1]
2015-11-02 14:53:04,792 INFO  stdout [[ Make sure atmosphere-compat-jboss.jar is not under your WEB-INF/lib and You must use the atmosphere-native-runtime dependency in order to use native Comet Support ]] [http-127.0.0.1/127.0.0.1:8080-1]
2015-11-02 14:53:04,792 INFO  stdout [[ there is no context.xml under WEB-INF Is the NIO or APR Connector enabled?]
 ]] [http-127.0.0.1/127.0.0.1:8080-1]
[AtmosphereFramework] [If you have more than one Connector enabled, make sure they both use the same protocol, e.g NIO/APR or HTTP for all. If not, org.atmosphere.container.BlockingIOCometSupport will be used and cannot be changed.]
 ]] [http-127.0.0.1/127.0.0.1:8080-1]
2015-11-02 14:53:04,798 INFO  stdout [[ java.lang.IllegalStateException: JBoss failed to detect this is a Comet application because the APR Connector is not enabled. 
 ]] [http-127.0.0.1/127.0.0.1:8080-1]
2015-11-02 14:53:04,798 INFO  stdout [[ Make sure atmosphere-compat-jboss.jar is not under your WEB-INF/lib and You must use the atmosphere-native-runtime dependency in order to use native Comet Support
 ]] [http-127.0.0.1/127.0.0.1:8080-1]
2015-11-02 14:53:04,798 INFO  stdout [[ there is no context.xml under WEB-INF
 ]] [http-127.0.0.1/127.0.0.1:8080-1]
2015-11-02 14:53:04,799 INFO  stdout [[     at org.atmosphere.container.JBossWebCometSupport.<clinit>(JBossWebCometSupport.java:51)
 ]] [http-127.0.0.1/127.0.0.1:8080-1]
2015-11-02 14:53:04,799 INFO  stdout [[     at sun.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
 ]] [http-127.0.0.1/127.0.0.1:8080-1]
2015-11-02 14:53:04,799 INFO  stdout [[     at sun.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:57)
 ]] [http-127.0.0.1/127.0.0.1:8080-1]
2015-11-02 14:53:04,799 INFO  stdout [[     at sun.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
 ]] [http-127.0.0.1/127.0.0.1:8080-1]
2015-11-02 14:53:04,799 INFO  stdout [[     at java.lang.reflect.Constructor.newInstance(Constructor.java:526)
 ]] [http-127.0.0.1/127.0.0.1:8080-1]
2015-11-02 14:53:04,799 INFO  stdout [[     at org.atmosphere.cpr.DefaultAsyncSupportResolver.newCometSupport(DefaultAsyncSupportResolver.java:231)
 ]] [http-127.0.0.1/127.0.0.1:8080-1]
2015-11-02 14:53:04,799 INFO  stdout [[     at org.atmosphere.cpr.DefaultAsyncSupportResolver.resolveMultipleNativeSupportConflict(DefaultAsyncSupportResolver.java:332)
 ]] [http-127.0.0.1/127.0.0.1:8080-1]
2015-11-02 14:53:04,799 INFO  stdout [[     at org.atmosphere.cpr.DefaultAsyncSupportResolver.resolveNativeCometSupport(DefaultAsyncSupportResolver.java:316)
 ]] [http-127.0.0.1/127.0.0.1:8080-1]
2015-11-02 14:53:04,799 INFO  stdout [[     at org.atmosphere.cpr.DefaultAsyncSupportResolver.resolve(DefaultAsyncSupportResolver.java:295)
 ]] [http-127.0.0.1/127.0.0.1:8080-1]
2015-11-02 14:53:04,799 INFO  stdout [[     at org.atmosphere.cpr.AtmosphereFramework.autoDetectContainer(AtmosphereFramework.java:2070)
 ]] [http-127.0.0.1/127.0.0.1:8080-1]
2015-11-02 14:53:04,799 INFO  stdout [[     at org.atmosphere.cpr.AtmosphereFramework.init(AtmosphereFramework.java:912)
 ]] [http-127.0.0.1/127.0.0.1:8080-1]
2015-11-02 14:53:04,800 INFO  stdout [[     at org.atmosphere.cpr.AtmosphereFramework.init(AtmosphereFramework.java:836)
 ]] [http-127.0.0.1/127.0.0.1:8080-1]
2015-11-02 14:53:04,800 INFO  stdout [[     at org.primefaces.push.PushServlet.configureFramework(PushServlet.java:84)
 ]] [http-127.0.0.1/127.0.0.1:8080-1]
2015-11-02 14:53:04,800 INFO  stdout [[     at org.primefaces.push.PushServlet.configureFramework(PushServlet.java:36)
 ]] [http-127.0.0.1/127.0.0.1:8080-1]
2015-11-02 14:53:04,800 INFO  stdout [[     at org.atmosphere.cpr.AtmosphereServlet.init(AtmosphereServlet.java:74)`

I have no time to install APR, because of high risks and BlockingIOCometSupport  is ok for me. I would like to set this parameter in web.xml to avoid the exception above.

**Comment by jfarcand** [comment]
@rustock0 Pull request welcomed :-) Remember than using  `BlockingIOCometSupport` is far from being recommended.

**Comment by miron4dev** [comment]
@jfarcand I will try prepare pull request during this week ;) Anyway, why `BlockingIOCometSupport` is not recommended? 

From javadoc:
`request thread will be blocked until another Thread invoke the {@link Broadcaster#broadcast}`
Does it mean that events published through EventBus will not be handled in parallel?

**Comment by jfarcand** [comment]
@rustock0 Yes to all. Blocking I/O is the worse case scenario and won't scale...henre the error message :-)

**Comment by miron4dev** [comment]
@jfarcand Thank you for the explanation. I will try use non-blocking I/O in near future, but now I should use what I have.
As for pull request, it requires more time than I expected. After research I found that `asyncSupport` is initialized before `endpointMapper`. So, in the constructor of `AsynchronousProcessor`

``` java
config.framework().endPointMapper()
```

returns `DefaultEndpointMapper`, but should `PushEndpointMapper`.

Even if I define 

``` xml
        <init-param>
            <param-name>org.atmosphere.cpr.EndpointMapper</param-name>
            <param-value>org.primefaces.push.impl.PushEndpointMapper</param-value>
        </init-param>
        <init-param>
            <param-name>org.atmosphere.cpr.asyncSupport</param-name>
            <param-value>org.atmosphere.container.BlockingIOCometSupport</param-value>
        </init-param>
```

for my PushServlet, it is not working. 
Maybe you will solve the problem faster me. I'm not an expert in Atmosphere. But I will try again soon.

**Comment by tandraschko** [comment]
PrimeFaces Push will be deprecated in 6.2 and removed in 6.3 ( see https://www.primefaces.org/primefaces-6-2-roadmap/)
Therefor i will close this issue as won't fix. If you will provide a PR to fix this issue until 6.2 RC1, i can merge it.
